### PR TITLE
Move markdownlint and markdown-link-check to github actions

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,11 +28,11 @@ install-markdown-link-check:
...
 	@for f in $(ALL_DOCS); do $(MARKDOWN_LINK_CHECK) --quiet --config .markdown_link_check_config.json $$f; done
 
-.PHONY: install-markdown-lint
+.PHONY: install-markdownlint
 install-markdown-lint:
 	npm install -g markdownlint-cli
 
-.PHONY: markdown-lint
+.PHONY: markdownlint
 markdown-lint:
```
Comment (Oberon00) [inline Makefile:36]: The PHONY no longer matches the actual target name now.

```suggestion
markdownlint:
```

same with install above

**Review by arminru — APPROVED** [review]
I would put both in the same workflow/YAML to reduce maintenance effort when changing triggers or the like.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Any reason to not include `misspell` as well? Or is that planned in a follow-up? Just curious ;)

**Comment by bogdandrutu** [comment]
Will be a follow-up

File: Makefile (hunk 1 above)
Comment (Oberon00) [reply Makefile:36]: @bogdandrutu Why did you resolve this without fixing the issue?

**Pull request merged**